### PR TITLE
add meek support

### PR DESCRIPTION
**Pull request by mnbogner**
add support for meek proxy

(this will require an updated maven dependency for IEnvoyProxy)

### CODE REVIEW

File: android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt [hunk 1]
```diff
@@ -759,6 +769,74 @@ class NetworkIntentService : IntentService("NetworkIntentService") {
...
+        if (meekUrl.isNullOrEmpty() || meekFront.isNullOrEmpty() || meekTunnel.isNullOrEmpty()) {
+            Log.e(TAG, "some arguments required for meek service are missing")
+        } else {
+
+            // set the login params
+            val meekUserString = "url=" + meekUrl + ";front=" + meekFront
+            val nullCharString = "" + Char.MIN_VALUE
+
+            // additional hardcoded meek parameters
+            val logLevel = "DEBUG"
```
Comment (vitriolix) [inline android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt:812]: why is debug / unsafeLogging hard coded to on?

File: android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt [hunk 2]
```diff
@@ -759,6 +769,74 @@ class NetworkIntentService : IntentService("NetworkIntentService") {
...
+
+            Log.d(TAG, "meek service started at " + urlString)
+
+            meekUrls.add(urlString)
+
+            // method returns port immediately but service is not ready immediately
+            Log.d(TAG, "submit url after a short delay for starting meek")
+            ioScope.launch() {
+                Log.d(TAG, "start meek delay")
+                delay(SHORT_DELAY)
```
Comment (vitriolix) [inline android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt:827]: what if Meek isn't ready yet?

Comment (mnbogner) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt:827]: the connection to the local proxy will fail, but during testing it has been ready within 1 second.

File: android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt (hunk 1 above)
Comment (mnbogner) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt]: fixed.  now set to ERROR/false/false

File: android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt (hunk 2 above)
Comment (mnbogner) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt:827]: tried lowering SHORT_DELAY to 500ms but then snowflake failed.  would rather not have a different delay for every service and i'm not sure 500ms is a big gain.  can revisit this approach later but the main difficulty is that iEnvoyProxy is calling a binary and i'm not aware of any way to get the feedback from that we'd need for more precise timing.

Comment (vitriolix) [reply android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt:827]: 500 ms is a pretty big chunk of time in terms of ui response, but let's not change this too much now, follow on funding we should definintely focus energy on how to implement an exponential backoff

**Comment by vitriolix** [comment]
🚢

**Pull request merged**